test/scalacheck/scala/tools/nsc/scaladoc/HtmlFactoryTest.scala

```diff
@@ -537,16 +537,17 @@ object HtmlFactoryTest extends Properties("HtmlFactory") {
     val files = createTemplates("basic.scala")
     //println(files)
 
-    property("class") = files.get("com/example/p1/Clazz.html").exists { page =>
-      val html = toHtml(page)
-
-      property("implicit conversion") = html contains """<span class="modifier">implicit </span>"""
-
-      property("gt4s") = html contains """title="gt4s: $colon$colon""""
-
-      property("gt4s of a deprecated method") = html contains """title="gt4s: $colon$colon$colon$colon. Deprecated: """
-
-      true
+    // class
+    {
+      val html = files.get("com/example/p1/Clazz.html")
+        .map(page => { lazy val s = toHtml(page); () => s })
+      property("class") = html.map(_()).isDefined
+      def ifExists(op: String => Prop) = html.map(_()).fold(undecided)(op)
+      property("class:implicit conversion") =
+        ifExists(_ contains """<span class="modifier">implicit </span>""")
+      property("class:gt4s") = ifExists(_ contains """title="gt4s: $colon$colon"""")
+      property("class:gt4s of a deprecated method") =
+        ifExists(_ contains """title="gt4s: $colon$colon$colon$colon. Deprecated: """)
```

Comment:
How is this related to the PR?

Reply:
so! fun story: the call to `property` appends the new property to a `ListBuffer` containing all the properties. during execution, we iterate over that `ListBuffer`, and if you create a new property during that iteration (which happens if you nest properties, which is not officially supported anyway), it breaks